### DISCUSSION TITLE
Add native support for Vultr Kubernetes Engine (VKE)

### DISCUSSION THREAD

**Discussion by pedrosimao** [kubero-dev/kubero#181] [Ideas/Features]
Vultr is a cheap and reliable cloud provider, furthermore their Kubernetes Engine is completely free: https://www.vultr.com/pricing/#kubernetes-engine

For small projects, or solo developers, this can be a super cheap option to have a small Kubernetes cluster that can scale.
I am willing to help...

**Comment by mms-gianni** [comment]
Thats a good idea. The Api looks straight forward too. 
https://www.vultr.com/api/#operation/create-kubernetes-cluster

**Reply by pedrosimao** [reply]
True that! Seems super simple. Can you point me to the kubero files connecting to other cloud suppliers. I can take a look and see if I can try to contribute on the Vultr side, myself.

**Reply by pedrosimao** [reply]
I guess those are the ones: https://github.com/kubero-dev/kubero-cli/blob/main/cmd/installdigitalocean.go
Any other places?

**Reply by mms-gianni** [reply]
Exactly. 
You might also have look into the Scaleway version since it has some useful queries. 
https://github.com/kubero-dev/kubero-cli/blob/main/cmd/installscaleway.go 

I need to refactor this at some point with a proper interface.

**Comment by mms-gianni** [comment]
Just remembered I had the same Idea in December 👍 
https://github.com/orgs/kubero-dev/projects/1/views/3?filterQuery=vul&pane=issue&itemId=16831451

**Reply by pedrosimao** [reply]
hahaha, humans have the same good ideas by accident. I remember once I lost a contest for someone who had a very similar idea to mine. So I got a second place. Anyway, maybe one extra reason to do it.

**Comment by secondmanveran** [comment]
@mms-gianni 

Is there any update on this? I use Vultr exclusively so I'd love to try out Kubero there.

Thanks

**Reply by mms-gianni** [reply]
@secondmanveran It should work there too. 

Just create a cluster, and run the kubero cli installer. Skip the creation part. Make sure you have kubectl installed and selected the desired cluster in your context.

![image](https://github.com/user-attachments/assets/72039cdd-ed0a-479c-bcea-9506e599b7aa) 

Kubero is currently limited to Nginx (planned to change that), so make sure the Nginx ingress is installed: 
https://docs.vultr.com/how-to-set-up-nginx-ingress-controller-with-ssl-on-vke 

It is still planned to add Vultr, but I currently got stuck in migrating to Kubero v3.

**Reply by hongkongkiwi** [reply]
When I pick Install Ingress Controller it doesn't give me an option of Vultr. Should I skip ingress?

It would be great to pick Vultr as it's one of the cheapest providers who also provide the control plane for free and they are reliable and well funded.

Linode is a bit similar but they are really difficult to sign up with and not very helpful if you fail their obscure process.
